### PR TITLE
Mice: Resting and squeaking

### PR DESCRIPTION
**Pull request by NanakoAC**
A PR for mice
Adds a resting animation, several new audio files, verbs to use them,  and a variety of audio changes.
Also fixes a bug where speech_chance was set to zero while sleeping, but not set back when the mouse wakes up

  - rscadd: "Mice now have a sprite for resting"
  - rscadd: "Added a few squeak verbs for mice with new audio"
  - tweak: "Mice will now occasionally squeak, and squeak chance increased"
  - tweak: "Mice will now squeal in pain when killed, and sometimes when stepped on"
  - bugfix: "Fixed a bug where mice would permanantly stop squeaking after sleeping once"

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 1]
```diff
@@ -95,12 +108,49 @@
...
+
+
+//Plays a random selection of four sounds, at a low volume
+//This is triggered randomly periodically by any mouse, or manually
+/mob/living/simple_animal/mouse/proc/squeak_soft()
+	set name = "Soft Squeaking"
+	set category = "Abilities"
+
+	var/sound = pick(soft_squeaks)
+	while (sound == last_softsqueak)
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: Why use a while() loop, and not do this:
```
var/list/new_squeaks = sound ? soft_squeaks - sound : soft_squeaks
sound = pick(new_squeaks)
```

Seems like it'd be more efficient that way?

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 2]
```diff
@@ -33,9 +39,9 @@
 
 /mob/living/simple_animal/mouse/Life()
 	..()
-	if(!stat && prob(speak_chance))
```
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: Won't this generate the chance of sleeping or dead mice squeaking?

Comment (NanakoAC) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: Sleeping no, but dead mice possibly. That would have technically allowed a mouse to still squeek if its controlled by a player AND dead, AND the player hasn't left the corpse as a ghost yet. A kinda minor edge case, but i'm fixing it anyway

**Comment by NanakoAC** [comment]
Changes made, this should be ready to go

File: code/modules/mob/living/simple_animal/friendly/mouse.dm (hunk 2 above)
Comment (skull132) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: Diid you fix this?

Comment (NanakoAC) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: yes fixed

**Pull request merged**